pkg/reconciler/configmappropagation/configmappropagation_test.go

```diff
@@ -0,0 +1,493 @@
+					WithConfigMapOwnerReference(NewConfigMapPropagation(configMapPropagationName, currentNS,
+						WithInitConfigMapPropagationConditions,
+						WithConfigMapPropagationSelector(selector),
+					)),
+				),
+			}},
+			WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+				Object: NewConfigMapPropagation(configMapPropagationName, currentNS,
+					WithInitConfigMapPropagationConditions,
+					WithConfigMapPropagationSelector(selector),
+					WithConfigMapPropagationPropagated,
+					WithInitConfigMapStatus(),
+					WithCopyConfigMapStatus("test-cmp-test-original-cm", "knative-eventing/test-original-cm",
+						"Copy", "True", ""),
+				),
+			}},
+			WantEvents: []string{
+				Eventf(corev1.EventTypeNormal, configMapPropagationPropagateSingleConfigMapSucceed, "Propagate ConfigMap test-original-cm succeed"),
+				Eventf(corev1.EventTypeNormal, configMapPropagationReadinessChanged, "ConfigMapPropagation %q became ready", configMapPropagationName),
+			},
+		}, {
+			Name: "Copy ConfigMap has changed",
+			Key:  testKey,
+			Objects: []runtime.Object{
+				NewConfigMapPropagation(configMapPropagationName, currentNS,
+					WithInitConfigMapPropagationConditions,
+					WithConfigMapPropagationSelector(selector),
+					WithInitConfigMapStatus(),
+				),
+				NewConfigMap(originalConfigMapName, originalNS,
+					WithConfigMapLabels(originalSelector),
+					WithConfigMapData(originalData),
+				),
+				NewConfigMap(copyConfigMapName, currentNS,
+					WithConfigMapLabels(copySelector),
+					WithConfigMapData(copyData),
+					WithConfigMapOwnerReference(NewConfigMapPropagation(configMapPropagationName, currentNS,
+						WithInitConfigMapPropagationConditions,
+						WithConfigMapPropagationSelector(selector),
+					)),
+				),
+			},
+			WantUpdates: []clientgotesting.UpdateActionImpl{{
+				Object: NewConfigMap(copyConfigMapName, currentNS,
+					WithConfigMapLabels(copySelector),
+					WithConfigMapData(originalData),
+					WithConfigMapOwnerReference(NewConfigMapPropagation(configMapPropagationName, currentNS,
+						WithInitConfigMapPropagationConditions,
+						WithConfigMapPropagationSelector(selector),
+					)),
+				),
+			}},
+			WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+				Object: NewConfigMapPropagation(configMapPropagationName, currentNS,
+					WithInitConfigMapPropagationConditions,
+					WithConfigMapPropagationSelector(selector),
+					WithConfigMapPropagationPropagated,
+					WithInitConfigMapStatus(),
+					WithCopyConfigMapStatus("test-cmp-test-original-cm", "knative-eventing/test-original-cm",
+						"Copy", "True", ""),
+				),
+			}},
+			WantEvents: []string{
+				Eventf(corev1.EventTypeNormal, configMapPropagationPropagateSingleConfigMapSucceed, "Propagate ConfigMap test-original-cm succeed"),
+				Eventf(corev1.EventTypeNormal, configMapPropagationReadinessChanged, "ConfigMapPropagation %q became ready", configMapPropagationName),
+			},
+		}, {
+			Name: "Copy ConfigMap no longer has required labels",
```

Comment:
Related to the comment above and #2437: if we write code to only delete the CMP owner reference, we should add a test case for abandoning a ConfigMap with multiple owner references.